src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final String defaultColor = "WHITE";
+    private final String defaultName = "circle";
+    private final int defaultRadius = 10;
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Figure[] figures = new Figure[]{new Circle(), new Rectangle(),
+                new IsoscelesTrapezoid(), new Square(), new RightTriangle()};
+        int randIndex = random.nextInt(figures.length);
```

Comment:
Here you create all figures in an array, for each method call. This is redundant. 
Instead you should generate the random values, create a random figure and pass previously generated random data to the constructor.